aarch64/src/runtime.rs

```diff
@@ -2,11 +2,30 @@
 
 extern crate alloc;
 
+use crate::uartmini::MiniUart;
 use alloc::alloc::{GlobalAlloc, Layout};
+use core::fmt::Write;
 use core::panic::PanicInfo;
+use port::devcons::EarlyConsole;
 
+// TODO
+//  - Add qemu integration test
+//  - Use LockingConsole via println!() macro once available
+//  - Add support for raspi4
 #[panic_handler]
-pub extern "C" fn panic(_info: &PanicInfo) -> ! {
+pub extern "C" fn panic(info: &PanicInfo) -> ! {
+    // Miniuart settings for raspi4 once mapped to higher half
+    //let uart = MiniUart::from_addresses(0xffff800000200000, 0xffff800000215000, 0xffff800000215040);
+
+    // Miniuart settings for raspi3 physical memory, as used by qemu currently
+    let uart = MiniUart::from_addresses(0x3f200000, 0x3f215000, 0x3f215040);
```

Comment:
When I did (essentially) this for Hypatia, I just doubled up the macros and wrote a `panic_println!()` and `panic_print!()` that did the usual things, but didn't lock; those could then be called directly from a panic context. They just made a UART device object internally, and dropped it when they were done.

Reply:
That's a little awkward with our setup, since we have the println macros in the port crate, and the uart in the arch crate.  So for now I'm going to leave it as it is.  I like the idea of a set of panic_print macros though - maybe we can get it working later.